### PR TITLE
Bug 1902076: Checking for route status

### PR DESCRIPTION
**Pull request by ricardomaraschini**
This commits adds a check for route statuses during operator status sync
process. If any of the routes was not Admitted the operator will report
itself as Degraded.

### CODE REVIEW

File: pkg/operator/controller.go [hunk 1]
```diff
@@ -248,7 +271,11 @@ func (c *Controller) sync() error {
 		deploy = deploy.DeepCopy() // make sure we won't corrupt the cached vesrion
 	}
 
-	c.syncStatus(cr, deploy, applyError)
+	routes, err := c.getRoutes(cr)
+	if err != nil {
+		return err
```
Comment (dmage) [inline pkg/operator/controller.go:276]: Failure to access one of the operator's routes shouldn't prevent the operator from reporting its status. Routes are expected to be absent when the operator is Removed (this blocks tests).

**Comment by dmage** [comment]
/bugzilla refresh
I'd prefer to be on the safe side and don't want to fix it just before code freeze.

File: pkg/operator/controller.go [hunk 2]
```diff
@@ -211,6 +212,29 @@ func (c *Controller) createOrUpdateResources(cr *imageregistryv1.Config) error {
 	return nil
 }
 
+// getRoutes returns a list of all routes configured for the image registry, including
+// the default route if configured.
+func (c *Controller) getRoutes(cr *imageregistryv1.Config) ([]*routev1.Route, error) {
+	var routes []*routev1.Route
+	if cr.Spec.DefaultRoute {
+		if route, err := c.listers.Routes.Get(defaults.RouteName); err != nil {
+			klog.V(5).Infof("unable to get default route: %s", err)
```
Comment (dmage) [inline pkg/operator/controller.go:221]: V(5) won't be visible even if you enable Debug logging. Only Trace or TraceAll that also includes HTTP requests from client-go.

File: pkg/operator/controller.go (hunk 1 above)
Comment (ricardomaraschini) [reply pkg/operator/controller.go:277]: done, ptal.

File: pkg/operator/controller.go (hunk 2 above)
Comment (ricardomaraschini) [reply pkg/operator/controller.go]: done, ptal.

**Pull request merged**